packages/vsce/src/utils/profileManagement.ts

```diff
@@ -312,21 +312,29 @@ export class ProfileManagement {
     group?: string
   ): Promise<{ cacheToken: string; recordCount: number; }> {
     const session = this.getSessionFromProfile(profile.profile);
-    const allProgramsResponse = await getResource(session, {
-      name: resourceName,
-      cicsPlex: plexName,
-      ...group ? { regionName: group } : {},
-      criteria: criteria,
-      queryParams: {
-        summonly: true,
-        nodiscard: true,
-        overrideWarningCount: true,
+    try {
+      const allProgramsResponse = await getResource(session, {
+        name: resourceName,
+        cicsPlex: plexName,
+        ...group ? { regionName: group } : {},
+        criteria: criteria,
+        queryParams: {
+          summonly: true,
+          nodiscard: true,
+          overrideWarningCount: true,
+        }
+      });
+      if (allProgramsResponse.response.resultsummary.api_response1_alt === "OK") {
+        if (allProgramsResponse.response && allProgramsResponse.response.resultsummary) {
+          const resultsSummary = allProgramsResponse.response.resultsummary;
+          return { cacheToken: resultsSummary.cachetoken, recordCount: parseInt(resultsSummary.recordcount, 10) };
+        }
       }
-    });
-    if (allProgramsResponse.response.resultsummary.api_response1_alt === "OK") {
-      if (allProgramsResponse.response && allProgramsResponse.response.resultsummary) {
-        const resultsSummary = allProgramsResponse.response.resultsummary;
-        return { cacheToken: resultsSummary.cachetoken, recordCount: parseInt(resultsSummary.recordcount, 10) };
+    } catch (error) {
+      if (error instanceof imperative.ImperativeError) {
+        if (!error.mDetails.msg.toUpperCase().includes("NODATA")) {
```

Comment:
I like the simple implementation.
I'm curious if we can look for something a bit more deterministic than just `NODATA`
Is there a specific HTTP code, or errorCode coming from the CMCI REST API?

Reply:
Disregard 😅 

---

After reading a bit more doc, NODATA seems to be standard for the CMCI APIs 😅

Reply:
Sad but true! 👍👍👎👎